apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -161,6 +160,38 @@ def _set_reader_coord(
             return True
         return False
 
+    @classmethod
+    def _dim_capacity_and_extent(
+        cls,
+        dim_name: str,
+        dim_shape: Optional[int],
+        create_options: TileDBCreateOptions,
+    ) -> Tuple[int, int]:
+        """
+        Given a user-specified shape (maybe ``None``) along a particular dimension,
+        returns a tuple of the TileDB capacity and extent for that dimension, suitable
+        for schema creation. If the user-specified shape is None, the largest possible
+        int64 is returned for the capacity.
+        """
+        if dim_shape is not None:
```

Comment:
maybe put the `dim_shape is None` case first to reduce the number of negations in the `if`

Reply:
done